test/Test/Pos/Block/Logic/Mode.hs

```diff
@@ -138,36 +141,67 @@ instance Buildable TestParams where
+-- misconfigured and it should be fixed immediately.
+-- | Map each participant coin in the distribution.
+stakeDistrAddCoin :: Coin -> StakeDistribution -> StakeDistribution
+stakeDistrAddCoin a d = case d of
+    (FlatStakes n c)       -> FlatStakes n (addMul c n)
+    (BitcoinStakes n c)    -> BitcoinStakes n (addMul c n)
+    (CustomStakes cs)      -> CustomStakes $ map (`unsafeAddCoin` a) cs
+    r@RichPoorStakes{..}   -> r { sdRichStake = addMul sdRichStake sdRichmen
+                                , sdPoorStake = addMul sdPoorStake sdPoor
+                                }
+    -- there's no way to make the fair mapping of exp stakes since
+    -- it's defined above and 'ExponentialStakes' doesn't contain all
+    -- the data needed to describe distr in full.
+    ExponentialStakes n mc -> ExponentialStakes n (a `unsafeAddCoin` mc)
+  where
+    addMul :: Coin -> Word -> Coin
+    addMul c n = c `unsafeAddCoin` (a `unsafeMulCoin` n)
+
 -- More distributions can be added if we want (e. g. RichPoor).
-genSuitableStakeDistribution :: Word -> Gen StakeDistribution
-genSuitableStakeDistribution stakeholdersNum =
-    oneof [genFlat{-, genBitcoin-}, pure (ExponentialStakes stakeholdersNum)]
+genSuitableStakeDistribution :: Coin -> Word -> Gen StakeDistribution
+genSuitableStakeDistribution dustThd stakeholdersNum =
+    -- Minimum coin in distribution should be equal or greater than
+    -- boot dust threshold.
+    --
+    -- Notice: it's only a _balance_ distribution, the real stake
+    -- distribution in boot era is defined by the boot stakeholders
+    -- set.
+    stakeDistrAddCoin dustThd <$>
+    oneof [ genFlat
+          {-, genBitcoin-} -- is broken
+          , pure (ExponentialStakes stakeholdersNum (mkCoin 1))]
   where
     totalCoins = mkCoin <$> choose (fromIntegral stakeholdersNum, unsafeGetCoin maxBound)
-    genFlat =
-        FlatStakes stakeholdersNum <$> totalCoins
-    -- Apparently bitcoin distribution is broken (it produces total stake
-    -- greater than requested one) and I don't think it's worth fixing it.
-    -- genBitcoin = BitcoinStakes stakeholdersNum <$> totalCoins
+    genFlat = FlatStakes stakeholdersNum <$> totalCoins
 
 instance Arbitrary TestParams where
     arbitrary = do
         secretKeysList <- toList @(NonEmpty SecretKey) <$> arbitrary -- might have repetitions
         let _tpStartTime = fromMicroseconds 0
         let invSecretsMap = mkInvSecretsMap secretKeysList
         let _tpAllSecrets = AllSecrets invSecretsMap
-        stakeDistribution <-
-            genSuitableStakeDistribution (fromIntegral $ length invSecretsMap)
         let addresses =
                 map (makePubKeyAddress . toPublic) (toList invSecretsMap)
-        let utxo = genesisUtxo Nothing [(addresses, stakeDistribution)]
-        let _tpGenTxpContext = mkGenesisTxpContext utxo
-        return TestParams {_tpStakeDistributions = one stakeDistribution, ..}
+        let genStakeholders =
```

Comment:
Why don't we use `genWStakeholders` here? It would add more versatility to tests.